### PR TITLE
Update Cypress to version 10

### LINKED ISSUE #6145: Update Cypress to version 10
**Detailed Description**
Update Cypress to latest version 10.

**Context**
Cypress reached recently version 10, which provides a better configuration,  extended from JSON to JS and even TS (most relevant feature for us),  and a new interface for the dashboard.
It also provides better component testing, which may be interesting to try it out.

The migration process is quite straightforward, but it requires to update some configuration, like exclusion of the setup, paths, linting and maybe more.

https://cypress.io/blog/2022/06/01/cypress-10-release/
https://docs.cypress.io/guides/references/migration-guide#Configuration-File-Changes

### PR DESCRIPTION
**Pull request by cnotv**
### Summary
Fixes #6145
Update Cypress to version 10

### Occurred changes and/or fixed issues
Please refer to the Cypress official page for new changes.

### Technical notes summary
- Fixed issue with duplicated tests while skipping
- Moved env assignment, configuration and test list from `index.ts` to `cypress.config.ts`
- Excluded Cypress folder from build/test tsconfig
- With Cypress 10 the skipped tests are now called `pending` 🤷 

### Areas or cases that should be tested
- Start project
- Start test

### Areas which could experience regressions
- Linting issues

### Screenshot/Video
Refer to actions below.

### CODE REVIEW

File: cypress.config.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+ */
+const getSpecPattern = (): string[] => {
+  const optionalPaths = [
+    {
+      path:   'cypress/e2e/tests/setup/**/*.spec.ts',
+      active: process.env.TEST_SKIP_SETUP !== 'true'
+    }
+  ];
+  const activePaths = optionalPaths.filter(({ active }) => Boolean(active)).map(({ path }) => path);
+  const paths = [
```
Comment (Shavindra) [inline cypress.config.ts:16]: You can just have the return statement here.

File: cypress.config.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+ * @returns
+ */
+const getSpecPattern = (): string[] => {
+  const optionalPaths = [
+    {
+      path:   'cypress/e2e/tests/setup/**/*.spec.ts',
+      active: process.env.TEST_SKIP_SETUP !== 'true'
+    }
+  ];
+  const activePaths = optionalPaths.filter(({ active }) => Boolean(active)).map(({ path }) => path);
```
Comment (Shavindra) [inline cypress.config.ts:15]: This could be just reduce instead of 2 array operations.

File: cypress.config.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
+import { defineConfig } from 'cypress';
+require('dotenv').config();
+
+/**
+ * Filter test spec paths based on env var configuration
+ * @returns
+ */
+const getSpecPattern = (): string[] => {
```
Comment (Shavindra) [inline cypress.config.ts:8]: Why does this need to be a function?

File: cypress.config.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    bootstrapPassword: process.env.CATTLE_BOOTSTRAP_PASSWORD,
+  },
+  e2e:                   {
+    // We've imported your old cypress plugins here.
+    // You may want to clean this up later by importing these.
+    setupNodeEvents(on, config) {
+      return require('./cypress/plugins/index.ts')(on, config);
+    },
+    experimentalSessionAndOrigin: true,
+    specPattern:                  getSpecPattern(),
```
Comment (Shavindra) [inline cypress.config.ts:42]: `getSpecPattern` can be just `const specPattern`?

File: cypress.config.ts (hunk 2 above)
Comment (cnotv) [reply cypress.config.ts:15]: Readability is preferred over smaller loops.

**Review by Shavindra — APPROVED** [review]

File: cypress.config.ts (hunk 3 above)
Comment (cnotv) [reply cypress.config.ts:8]: Because there's a bunch of computations and there's no need to keep them exposed outside the function.

**Pull request merged**